### ISSUE TITLE
Implement a Leveling System based on XP

### ISSUE THREAD

**Issue by RazvanBerbece** [RazvanBerbece/Aztebot#1]
# Story
One of the core requirements of the bot application is to enroll and incentivise community members in its leveling system based on experience points. The levels drive the roles on the server, and the roles drive the content and features available to a member of the community. Experience is gained by contributing to the server, helping others, coming up with useful posts, etc.. 

# Requirements
- [x] Figure out the levels -- are they numerical (1-10?), are they categorical (bad examples but examples nonetheless: `aztec inferior`, `aztec superior`?), do they have an upper limit (level 100?, `aztec profet lol`?)
- [x] Figure out the experience points gaining system -- how do we quantify experience gains (number of reactions?, some cool sentiment analysis for member messages?, endorsements? each like=1 xp point?), how do we quantify progression (how many points? how much XP? till next level)
- [ ] Link levels to the Discord roles -- we might need constant pollers of the channels (implies additional infrastructure) to check for contributions and process them into XP for the owners?
- [ ] Unknown unknowns?

# A few wild ideas
1. Maybe the XP and leveling system could be implemented through some sort of voting system?

Note: This story is a work in progress and the design of the system may change over time. Come back here for design or technical updates.

**Comment by RazvanBerbece** [comment]
# Update 14 Mar 2024
As it stands, there is now currently a way of quantifying and putting members on a progression framework through:
- having XP gains associated to member actions like sending messages, spending time in VCs, etc.
- total XP is available for each member at any time

What's left to fully implement the automated progression system is:
- create a code-based leveling framework using the existing inner circle roles and a calculation on how much XP is needed for each role

**Comment by RazvanBerbece** [comment]
# Update 14 may 2024

Implemented in #62 

```
First Order  - 7.500XP // 15 ORE PE VOICE //  1.000 MESAJE TRIMISE

Zelator - 7.500XP // 15 ORE PE VOICE //  1.000 MESAJE TRIMISE
Theoricus - 10.000XP // 20 ORE PE VOICE //  2.500 MESAJE TRIMISE
Philosophus - 15.000XP // 30 ORE PE VOICE // 5.000 MESAJE TRIMISE

 Second Order  25.000XP // 40 ORE PE VOICE // 12.500 MESAJE TRIMISE

Adeptus Minor - 25.000XP // 40 ORE PE VOICE // 12.500 MESAJE TRIMISE
Adeptus Major - 35.000XP // 45 ORE PE VOICE // 15.000 MESAJE TRIMISE
Adeptus Exemptus - 50.000XP // 50 ORE PE VOICE // 20.000 MESAJE TRIMISE // PARTICIPARE LA MACAR 5 EVENIMENTE

 Third Order 100.000XP // 200 ORE PE VOICE // 35.000 MESAJE TRIMISE

Magister Templi - 100.000XP // 200 ORE PE VOICE // 35.000 MESAJE TRIMISE
Magus - 150.000XP // 250 ORE PE VOICE // 45.000 MESAJE TRIMISE
Ipsississimus - 200.000XP // 300 ORE PE VOICE // 50.000 MESAJE TRIMISE 
```
